modules/web-demo/src/components/Coins/coinFactory.ts

```diff
@@ -0,0 +1,196 @@
+import { BitGoAPI } from '@bitgo/sdk-api';
+import { BaseCoin } from '@bitgo/sdk-core';
+
+class CoinFactory {
+  private coinCache: Map<string, BaseCoin>;
+
+  constructor() {
+    this.coinCache = new Map();
+  }
+
+  private cacheAndRegister(
+    name: string,
+    sdk: BitGoAPI,
+    register: (sdk: BitGoAPI) => void,
+  ) {
+    register(sdk);
+    const coin = sdk.coin(name);
+    this.coinCache.set(name, coin);
+    return coin;
+  }
+
+  async getCoin(name: string, sdk: BitGoAPI): Promise<BaseCoin | undefined> {
```

Comment:
shouldn't this always return a BaseCoin? if not, it should  throw an error or something, no?

```suggestion
  async getCoin(name: string, sdk: BitGoAPI): Promise<BaseCoin> {
```